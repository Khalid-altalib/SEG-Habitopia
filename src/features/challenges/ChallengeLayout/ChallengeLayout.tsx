import { Box, Center, Heading, ScrollView, VStack } from "native-base";
import React from "react";
import { View, FlatList, Text } from "react-native";
import RegularLayout from "../../../components/RegularLayout/RegularLayout";
import styles from "../../../constants/Styles";
import ChallengeBox from "../ChallengeBox/challengeBox";
import ChallengeBoxes from "../ChallengeBoxes/challengeBoxes";

type Props = {
  children: React.ReactNode;
};

function ChallengeLayout(props: Props) {
  const { children } = props;
  return (
    <ScrollView>
      <RegularLayout>
        <Center>
          <Heading mb={4}>Pick a category ðŸš€</Heading>
        </Center>

        <VStack space={3} width={"100%"}>
          {children}
        </VStack>
      </RegularLayout>
    </ScrollView>
  );
}

export default ChallengeLayout;
