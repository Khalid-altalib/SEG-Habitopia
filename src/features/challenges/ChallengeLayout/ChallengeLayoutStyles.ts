import { background, border } from "native-base/lib/typescript/theme/styled-system";
import { StyleSheet } from "react-native";

const challengeLayoutStyles = StyleSheet.create({
challengeTitle: {
    fontSize: 30,
},
});

export default challengeLayoutStyles;