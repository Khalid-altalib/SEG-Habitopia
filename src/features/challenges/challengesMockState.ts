import { ChallengesState } from "./challengesSlice";

// Mocks the challenge state for testing purposes
const challengesMockState: ChallengesState = {
    challenges: [
        {
            id: '1',
            name: 'Fitness',
            description: 'Be active',
            active: true

        },
        {
            id: '1',
            name: 'Study',
            description: 'Study hard',
            active: false
        }
    ],
    fetchChallenges: {
        loading: false,
        error: ""
    },
    joinChallenge: {
        loading: false,
        error: ""
    }
};

export default challengesMockState;