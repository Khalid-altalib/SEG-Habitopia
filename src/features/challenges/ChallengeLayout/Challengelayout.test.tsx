// Importing necessary dependencies/modules/packages for the test file
import { render } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import challengesMockState from "../challengesMockState";
import ChallengeLayout from "./ChallengeLayout";

describe("ChallengeLayout", () => {
  const mockState = {
    // create a mock state for the Redux store
    challenges: challengesMockState,
  };

  // create a mock Redux store with the mock state and middleware
  const mockStore = configureStore([thunk])(mockState);

  it("renders the challenge layout correctly", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <ChallengeLayout>
          <></>
        </ChallengeLayout>
      </TestingWrapperNavigation>
    );

    // expect the rendered component to match the snapshot
    expect(component).toMatchSnapshot();
  });
});
