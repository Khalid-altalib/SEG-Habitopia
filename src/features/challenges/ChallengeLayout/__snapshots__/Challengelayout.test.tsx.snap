// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeLayout renders the challenge layout correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                [
                  {},
                  {
                    "dataSet": {},
                  },
                  {},
                ]
              }
              dataSet={{}}
              style={{}}
            >
              <View>
                <View
                  style={
                    {
                      "height": "100%",
                      "padding": 25,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={
                      {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "center",
                        "marginBottom": 25,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "fontFamily": undefined,
                            "fontSize": 23,
                          },
                          undefined,
                        ]
                      }
                    >
                      Pick a Challenge! ðŸš€
                    </Text>
                  </View>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "width": "100%",
                      }
                    }
                  />
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
